src/UserService.Business/Commands/Certificate/RemoveCertificateCommand.cs

```diff
@@ -1,56 +1,111 @@
 using LT.DigitalOffice.Kernel.Extensions;
 using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.Models.Broker.Enums;
+using LT.DigitalOffice.Models.Broker.Requests.Image;
 using LT.DigitalOffice.UserService.Business.Commands.Certificate.Interfaces;
 using LT.DigitalOffice.UserService.Data.Interfaces;
 using LT.DigitalOffice.UserService.Models.Db;
+using MassTransit;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
 using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
 
 namespace LT.DigitalOffice.UserService.Business.Commands.Certificate
 {
-    public class RemoveCertificateCommand : IRemoveCertificateCommand
+  public class RemoveCertificateCommand : IRemoveCertificateCommand
+  {
+    private readonly IAccessValidator _accessValidator;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IUserRepository _userRepository;
+    private readonly ICertificateRepository _certificateRepository;
+    private readonly IRequestClient<IRemoveImagesRequest> _removeImagesRequest;
+    private readonly ILogger<RemoveCertificateCommand> _logger;
+
+    public RemoveCertificateCommand(
+      IAccessValidator accessValidator,
+      IHttpContextAccessor httpContextAccessor,
+      IUserRepository userRepository,
+      ICertificateRepository certificateRepository,
+      IRequestClient<IRemoveImagesRequest> removeImagesRequest,
+      ILogger<RemoveCertificateCommand> logger)
     {
-        private readonly IAccessValidator _accessValidator;
-        private readonly IHttpContextAccessor _httpContextAccessor;
-        private readonly IUserRepository _userRepository;
-        private readonly ICertificateRepository _certificateRepository;
-
-        public RemoveCertificateCommand(
-            IAccessValidator accessValidator,
-            IHttpContextAccessor httpContextAccessor,
-            IUserRepository userRepository,
-            ICertificateRepository certificateRepository)
-        {
-            _accessValidator = accessValidator;
-            _httpContextAccessor = httpContextAccessor;
-            _userRepository = userRepository;
-            _certificateRepository = certificateRepository;
-        }
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _userRepository = userRepository;
+      _certificateRepository = certificateRepository;
+      _removeImagesRequest = removeImagesRequest;
+      _logger = logger;
+    }
+
+    public async Task<OperationResultResponse<bool>> Execute(Guid certificateId)
+    {
+      OperationResultResponse<bool> result = new();
+      List<string> resultErrors = result.Errors;
+
+      Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+      DbUser sender = _userRepository.Get(senderId);
+      DbUserCertificate userCertificate = _certificateRepository.Get(certificateId);
+
+      if (!(sender.IsAdmin ||
```

Comment:
if you get certificate before this, better first check last condition and after it go to use broker

Reply:
++